pkg/metrics/shoot.go

```diff
@@ -81,9 +80,7 @@ func (c gardenMetricsCollector) collectShootMetrics(ch chan<- prometheus.Metric)
 		)
 		isSeed = usedAsSeed(shoot)
 
-		if shootPurpose, ok := shoot.Annotations[constants.GardenPurpose]; ok {
-			purpose = shootPurpose
-		}
+		purpose = string(*shoot.Spec.Purpose)
```

Comment:
Can you check if shoot.Spec.Purpose `!= nil` before setting it?

Reply:
done